modules/caddytls/connpolicy.go

```diff
@@ -461,6 +462,16 @@ func setDefaultTLSParams(cfg *tls.Config) {
 	cfg.PreferServerCipherSuites = true
 }
 
+// SSLKEYLOGFILE
+func setSSLKeylog(cfg *tls.Config) {
+	// TODO: Take sslkeylogfile as the file to write.
+	w, err := os.OpenFile("tls-secrets.txt", os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
```

Comment:
This whole function / approach will probably need to be revised. It's a good start, but it needs to be more flexible and configurable I think.

Reply:
Something that would be nice (that I don't know how to do), is to also pipe the key logs to the regular Caddy logs.

Maybe something like "include_sslkeylog_logs"?

This would mean the end-user has to extract out the keys from the caddyserver log, but it would be much more flexible than only writing to a single file. e.g. writing to a network socket.

Reply:
Sorry, I got busy with other Caddy projects and am now traveling to help a family move. I'll try to get back to this ASAP.

Here's a quick example of how to emit a log (you can get an instance of a logger, if you don't already have one, from the `caddy.Context` during `Provision`): https://github.com/caddyserver/caddy/blob/03b5debd958a324d5c00cc37f887130b0198e747/modules/caddypki/maintain.go#L91-L92